src/components/implementation/simple_mc_udp_server/simple_mc_udp_server_no_lwip/main.c

```diff
@@ -4,18 +4,79 @@
 #include <mc.h>
 #include <simple_udp_stack.h>
 #include <netshmem.h>
+#include <sched.h>
 
 static int fd;
+static volatile thdid_t init_thd = 0;
 
 void
 cos_init(void)
 {
 	shm_bm_objid_t objid;
 	struct netshmem_pkt_buf *obj;
-
+	init_thd = cos_thdid();
 	/* create current component's shmem */
 	netshmem_create();
+}
+void
+cos_parallel_init(coreid_t cid, int init_core, int ncores)
```

Comment:
This function looks quite odd. Should we discuss? Seems to assume a very specific composition script, with a very specific setup.